app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -268,7 +268,14 @@ abstract class BaseBrowserFragment : Fragment(), BackHandler, SessionManager.Obs
                     context = context,
                     didFail = downloadJobStatus == AbstractFetchDownloadService.DownloadJobStatus.FAILED,
                     download = download,
-                    tryAgain = downloadFeature::tryAgain
+                    tryAgain = downloadFeature::tryAgain,
+                    onCannotOpenFile = {
+                        FenixSnackbar.make(view, Snackbar.LENGTH_SHORT)
+                            .setText(context.getString(R.string.mozac_feature_downloads_could_not_open_file))
+                            .setAnchorView(browserToolbarView.view)
```

Comment:
No action needed here, but I'm concerned about `BaseBrowserFragment#initializeUI`.  It's already 255 lines!  We could pull these initializations out into helper methods, but it would mostly be a cosmetic change.

I don't know what to do here yet, but I think this is an important smell that we should be thinking about.